### ISSUE TITLE
[Bug]: (recovering comments failed due to "invalidFilters")

### ISSUE THREAD

**Issue by GaviLazan** [ThioJoe/YT-Spammer-Purge#810]
### Duplicate Issues

- [X] There are **no existing posts relating to my problem**
- [ ] There are **existing posts relating to my problem, but the solution given, doesn't work for me.**

### What happened?

When trying to recover deleted comments from a log file I get the following error:

Traceback (most recent call last):
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1674, in <module>
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1651, in main
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 902, in primaryInstance
  File "Scripts\prepare_modes.py", line 467, in recover_deleted_comments
  File "Scripts\operations.py", line 1196, in check_recovered_comments
  File "googleapiclient\_helpers.py", line 131, in positional_wrapper
  File "googleapiclient\http.py", line 937, in execute
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments returned "The request specifies an invalid filter parameter.". Details: "[{'message': 'The request specifies an invalid filter parameter.', 'domain': 'youtube.parameter', 'reason': 'invalidFilters', 'location': 'parameters.', 'locationType': 'other'}]">
------------------------------------------------
Error Message: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments returned "The request specifies an invalid filter parameter.". Details: "[{'message': 'The request specifies an invalid filter parameter.', 'domain': 'youtube.parameter', 'reason': 'invalidFilters', 'location': 'parameters.', 'locationType': 'other'}]">
Status Code: 400
    Reason: invalidFilters

Tried to recover a second time with a different log file and got the same error.

### Release version

2.16.9

### Steps to reproduce

1. recover deleted comments (8)
2. load log file
3. hit enter to begin recovery
4. fail

### What platform are you seeing this problem on?

Windows (.exe file)

### Relevant log output

```shell
Traceback (most recent call last):
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1674, in <module>
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 1651, in main
  File "D:\Users\Joe\Documents\GitHub ThioJoe\YouTube-Spammer-Purge\YTSpammerPurge.py", line 902, in primaryInstance
  File "Scripts\prepare_modes.py", line 467, in recover_deleted_comments
  File "Scripts\operations.py", line 1196, in check_recovered_comments
  File "googleapiclient\_helpers.py", line 131, in positional_wrapper
  File "googleapiclient\http.py", line 937, in execute
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments returned "The request specifies an invalid filter parameter.". Details: "[{'message': 'The request specifies an invalid filter parameter.', 'domain': 'youtube.parameter', 'reason': 'invalidFilters', 'location': 'parameters.', 'locationType': 'other'}]">
------------------------------------------------
Error Message: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments returned "The request specifies an invalid filter parameter.". Details: "[{'message': 'The request specifies an invalid filter parameter.', 'domain': 'youtube.parameter', 'reason': 'invalidFilters', 'location': 'parameters.', 'locationType': 'other'}]">
Status Code: 400
    Reason: invalidFilters
```

### Screenshots

![Screenshot 2022-04-27 133451](https://user-images.githubusercontent.com/73096923/165499784-8c7d7eb9-4cc9-4e71-a594-f359c46ed8b9.png)

**Comment by ThioJoe** [comment]
Ugh I guess they made another undocumented change in the API, I'll have to take a look

**Comment by ThioJoe** [comment]
Hm one thing I'm not sure about is why it says it loaded 158 comments, but then only goes to recover 2. Could you provide the log file you're using? You can just drag it into the message box to upload it

**Comment by GaviLazan** [comment]
I had done a scan that turned up a ton of results I wanted to exclude (I was doing a search for hate speech in comments) and only after going through each comment and marking down which should get removed did I realize I needed to use the second list lol
So I duplicated the log and thought I might be able to remove the ones I wanted to keep and then delete the rest. That still ended up deleting all the comments.
I tried recovering from that file and failed.
I tried from the original log and failed.
[hatecomments.txt](https://github.com/ThioJoe/YT-Spammer-Purge/files/8578453/hatecomments.txt)

**Comment by joegrand** [comment]
FWIW, I'm seeing a similar failure when following these same steps as @GaviLazan, though my error response is `missingRequiredParameter` instead of `invalidFilters`:

```  
File "/Users/joegrand/Library/Python/3.9/lib/python/site-packages/googleapiclient/http.py", line 937, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments?part=snippet&fields=items&textFormat=plainText&alt=json returned "No filter selected. Expected one of: parentId, id". Details: "[{'message': 'No filter selected. Expected one of: parentId, id', 'domain': 'youtube.parameter', 'reason': 'missingRequiredParameter', 'location': 'parameters.', 'locationType': 'other'}]">
------------------------------------------------
Error Message: <HttpError 400 when requesting https://youtube.googleapis.com/youtube/v3/comments?part=snippet&fields=items&textFormat=plainText&alt=json returned "No filter selected. Expected one of: parentId, id". Details: "[{'message': 'No filter selected. Expected one of: parentId, id', 'domain': 'youtube.parameter', 'reason': 'missingRequiredParameter', 'location': 'parameters.', 'locationType': 'other'}]">
Status Code: 400
    Reason: missingRequiredParameter
```

I believe this is a related issue, but I can open another issue if not. 

I have 24 comment IDs successfully loaded, but still get the same progress indicator showing recovering only 2:

`Recovering Comments... - Progress: [0 / 2] (In Groups of 50)`

My release version is also 2.16.9, though I'm on macOS 10.14.6. Full log and restore list (of comments I'm trying to recover) attached. 

Thanks for an amazing tool, BTW!

[Joe Log.txt](https://github.com/ThioJoe/YT-Spammer-Purge/files/8672815/Joe.Log.txt)
[Joe Restore.txt](https://github.com/ThioJoe/YT-Spammer-Purge/files/8672828/Joe.Restore.txt)

**Comment by ThioJoe** [comment]
This issue is really old but I figured out the problem. It was a problem with how the variable was being returned in one of the functions. Unfortunately it had basically broken the entire recovery functionality. But it should now be fixed in the next release.
